monitor_workflow.rb

```diff
@@ -1,12 +1,46 @@
 require_relative './config.rb'
 
 require 'resque-scheduler'
+require 'faraday'
+require 'time'
+
+QUERY_RESULT_LIMIT = 10
+
 
 class MonitorWorkflow
   @queue = :job
 
   def self.perform
-    $log.info 'Monitoring workflow'
-    $log.info 'MP will implement me'
+    $log.info 'Checking for recent workflow results'
+
+    conn = Faraday.new(url: DAP_BASE_URL, ssl: {verify: false})
+    resp = conn.get(
+        '/api/v1/threat_assessment_runs',
+        {
+            private_token: PRIVATE_TOKEN,
+            sort_by: :start_date,
+            sort_order: :desc,
+            status: :finished,
```

Comment:
I think that this question touches the threat_assessment_run's role matter, and might be unanswerable at this point in time.

If we assume threat_assessment_run is short lived, and there is one for each hypgen call, (in my opinion) this is the correct way to determine if the system is alive. Consider that Threat_assessment_run has no timeout, so it may be stuck in 'running' state forever. When this happens we don't have a reliable way to tell if it's still producing results or not (there is no timestamp on TA entity, we don't know if one is being produced or not etc.). Also this 'live or dead' detection doesn't aspire to know what failed, and to what extent, so the "OK" situation should be reserved only for cases where we are fully satisfied with the results. I think that the result we are looking for from our system is a complete TAR run in some recent time period, (this should be an effect of constant levee monitoring).

Also, consider that there are still some unknown details, like how fast should we know about system failure etc. I think that the current setup will be sufficient until we obtain further requirements.